buku

```diff
@@ -614,32 +615,82 @@ class BukuDb:
+                    # first iteration, just print the results and ask for confirmation
+                    count = 0
+                    for row in results:
+                        count += 1
+                        print_record(row, count)
+                    resp = input('Delete search results? Enter \x1b[1my\x1b[21m to confirm: ')
+                    if resp != 'y' and resp != 'Y':
+                        return
+                    # the results have been printed, let's delete them one by one
+                    ask = False
+                    continue
+                else:
+                    # delete the records now (no confirmation)
+                    # db will be compacted after each delete, so delete the first item and search again
+                    bdb.delete_bookmark(int(results[0][0]))
+            else:
+                print(format_json(results))
 
-    def search_by_tag(self, tag, json=False):
+    def search_by_tag(self, tag, json=False, delete=False):
         """Search and list bookmarks with a tag
 
         :param tag: tag to search
         :param json: print in json format
+        :param delete: delete search results
         """
 
-        self.cur.execute("SELECT id, url, metadata, tags, desc FROM bookmarks WHERE tags LIKE '%' || ? || '%'", (tag,))
-        results = self.cur.fetchall()
-        if len(results) == 0:
-            return
+        # if it's search & delete, ask for confirmation, but only once
+        ask = delete
 
-        if not json:
-            prompt(results, self.noninteractive)
-        else:
-            print(format_json(results))
+        while True:
+            self.cur.execute("SELECT id, url, metadata, tags, desc FROM bookmarks WHERE tags LIKE '%' || ? || '%'", (tag,))
+            results = self.cur.fetchall()
+            if len(results) == 0:
+                return
+
+            if not json:
+                if delete is False:
+                    # just search and open results if wanted
+                    prompt(results, self.noninteractive)
+                    return
+                # time to delete, now..
+                if ask is True:
+                    # first iteration, just print the results and ask for confirmation
+                    count = 0
+                    for row in results:
+                        count += 1
+                        print_record(row, count)
+                    resp = input('Delete search results? Enter \x1b[1my\x1b[21m to confirm: ')
+                    if resp != 'y'and resp != 'Y':
+                        return
+                    # the results have been printed, let's delete them one by one
+                    ask = False
+                    continue
+                else:
+                    # delete the records now (no confirmation)
+                    # db will be compacted after each delete, so delete the first item and search again
+                    bdb.delete_bookmark(int(results[0][0]))
```

Comment:
> The query should be fired after the user confirmation.
Actually, the first query is just to display the results, since the use needs to review first the bookmarks to be deleted..
Then, if the c nfirmation is ok, query & delele in a loop takes place (with no confirmation)
Probably I should have explained a little bit the logic..
The loop continues just for delete. For search-only, it's just one query